L1Trigger/L1TTrackMatch/plugins/L1TrackJetClustering.h

```diff
@@ -113,6 +113,76 @@ namespace l1ttrackjet {
     return PassQuality;
   }
 
+  // Eta binning following the hardware logic
+  inline unsigned int eta_bin_firmwareStyle(int eta_word) {
+    //Function that reads in 15-bit eta word (in two's complement) and returns the index of eta bin in which it belongs
+    //Logic follows exactly according to the firmware
+    //We will first determine if eta is pos/neg from the first bit. Each half of the grid is then split into four coarse bins. Bits 2&3 determine which coarse bin to assign
+    //The coarse bins are split into 5 fine bins, final 13 bits determine which of these coarse bins this track needs to assign
+
+    int eta_coarse = 0;
+    int eta_fine = 0;
+
+    if (eta_word & (1 << 15)) {  //If eta is negative (first/leftmost bit is 1)
+      //Second and third bits contain information about which of four coarse bins
+      eta_coarse = 5 * ((eta_word & (3 << 13)) >> 13);
```

Comment:
It might be more readable to use the ap datatype's ```range(...)``` rather than bit shift, mask, bit shift (carefully)

Reply:
We are not familiar with the range() function. The bit shift, mask, bitshift, is more identical to firmware but I agree that it's not the most readable. Worth changing for this?